src/app/gps/qgsappgpsconnection.cpp

```diff
@@ -178,11 +178,18 @@ void QgsAppGpsConnection::disconnectGps()
 
 void QgsAppGpsConnection::onTimeOut()
 {
-  disconnectGps();
+  std::unique_ptr< QgsGpsConnection > oldConnection( mConnection );  emit disconnected();
```

Comment:
Is the issue with calling disconnectGps that mConnection is a null pointer here? Otherwise this code is all identical to the code in disconnectGps...

Reply:
@nyalldawson , I am not very expert with C ++, mConnection is a Null YES, maybe you can write differently! I just try to unlock Qgis.

Reply:
Can you test https://github.com/qgis/QGIS/pull/51413 ? It's an alternative approach which should fix this.

Reply:
@nyalldawson  Test done. Everything OK. Fixes the two timeout events I had been experiencing!

Reply:
Thanks for the test @bettellam !